oc-chef-pedant/lib/pedant/core_ext/net_http.rb

```diff
@@ -13,85 +13,108 @@
         s = OpenSSL::SSL::SSLSocket.new(s, @ssl_context)
         s.sync_close = true
-      end
-      @socket = BufferedIO.new(s)
-      @socket.read_timeout = @read_timeout
-      @socket.continue_timeout = @continue_timeout
-      @socket.debug_output = @debug_output
-      if use_ssl?
-        begin
-          if proxy?
-            @socket.writeline sprintf('CONNECT %s:%s HTTP/%s',
-                                      @address, @port, HTTPVersion)
-            @socket.writeline host_header
-            if proxy_user
-              credential = ["#{proxy_user}:#{proxy_pass}"].pack('m')
-              credential.delete!("\r\n")
-              @socket.writeline "Proxy-Authorization: Basic #{credential}"
-            end
-            @socket.writeline ''
-            HTTPResponse.read_new(@socket).value
-          end
-          # Server Name Indication (SNI) RFC 3546
-          s.hostname = @address if s.respond_to? :'hostname='
-          Timeout.timeout(@open_timeout) { s.connect }
-          if @ssl_context.verify_mode != OpenSSL::SSL::VERIFY_NONE
-            s.post_connection_check(@address)
-          end
-        rescue => exception
-          D "Conn close because of connect error #{exception}"
-          @socket.close if @socket and not @socket.closed?
-          raise exception
+        # Server Name Indication (SNI) RFC 3546
+        s.hostname = @address if s.respond_to? :hostname=
+        if @ssl_session and
+           Process.clock_gettime(Process::CLOCK_REALTIME) < @ssl_session.time.to_f + @ssl_session.timeout
+          s.session = @ssl_session if @ssl_session
         end
+        ssl_socket_connect(s, @open_timeout)
+        if @ssl_context.verify_mode != OpenSSL::SSL::VERIFY_NONE
+          s.post_connection_check(@address)
+        end
+        @ssl_session = s.session
+        D "SSL established"
       end
+      @socket = BufferedIO.new(s, read_timeout: @read_timeout,
+                               continue_timeout: @continue_timeout,
+                               debug_output: @debug_output)
       on_connect
-    end
-
-    def host_header
-      if ipv6_address?(@address)
-        "Host: [#{@address}]:#{@port}"
-      else
-        "Host: #{@address}:#{@port}"
+    rescue => exception
+      if s
+        D "Conn close because of connect error #{exception}"
+        s.close
       end
+      raise
     end
+    private :connect 
 
-    # This does not verify that the address is a
-    # valid ipv6 address. Rather, it assumes that any
-    # address string that contains a colon is an
-    # ipv6 address. Since colons are not allowed in
-    # domain names, it should fail on a name lookup.
-    # By the time this is called, the socket should be
-    # open, so whatever address is being used would
-    # technically be a valid one.
-    def ipv6_address?(addr)
-      return true if addr =~ /:/
-      return false
+    def host_header		
+      if ipv6_address?(@address)		
+        "Host: [#{@address}]:#{@port}"		
```

Comment:
In line 49 above we also print Host: so this probably needs to be fixed so we don't print Host: Host: addr:port

Reply:
done!